reactor-core/src/main/java/reactor/core/publisher/Hooks.java

```diff
@@ -306,6 +306,35 @@ public static void onDiscard(Consumer<Object> c) {
 		}
 	}
 
+	/**
+	 * Create an adapter for local {@link #onDiscard(Consumer)} hooks that check the element
+	 * being discarded is of a given {@link Class}. The resulting {@link Function} adds the
+	 * hook to the {@link Context}, potentially chaining it to an existing hook in the {@link Context}.
+	 *
+	 * @param type the type of elements to take into account
+	 * @param discardHook the discarding handler for this type of elements
+	 * @param <R> element type
+	 * @return a {@link Function} that can be used to modify a {@link Context}, adding or
+	 * updating a context-local discard hook.
+	 */
+	static final <R> Function<Context, Context> discardLocalAdapter(Class<R> type, Consumer<? super R> discardHook) {
+		final Consumer<Object> safeConsumer = obj -> {
```

Comment:
Add null check (Objects.requireNotNull)

Reply:
I guess the second statement should be `Objects.requireNonNull(discardHook, "onDiscard must be provided a discardHook Consumer")`